File: 백준/Silver/2217. 로프/로프.py
import sys
input = lambda : sys.stdin.readline()

n = int(input())

arr = []

for _ in range(n):
    arr.append(int(input()))

arr.sort(reverse= True)

max_v = 0
sum = 0
for i in range(n):
    sum += arr[i]
    temp = sum // (i + 1)
    if arr[i] * (i + 1) > temp:
        max_v = max(arr[i] * (i + 1), max_v)
    else:
        max_v = max(sum // (i + 1), max_v)

print(max_v)

File: 백준/Silver/2217. 로프/로프.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.Collections;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws Exception{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		Integer n = Integer.parseInt(br.readLine());
		Integer[] arr = new Integer[n];
		
		for(int i = 0; i < n; i++)
			arr[i] = Integer.parseInt(br.readLine());

		Arrays.sort(arr, Collections.reverseOrder());
		int max = 0;
		int sum = 0;
		
		for(int i = 0; i < n; i ++) {
			sum += arr[i];
			int temp = sum / (i + 1);
			if (arr[i] * (i + 1) > temp)
				max = Math.max(max, arr[i] * (i + 1));
			else
				max = Math.max(max, temp);
		}
		
		System.out.println(max);

	}
}
